snuba/processor.py

```diff
@@ -53,7 +62,7 @@ class InvalidMessageVersion(Exception):
 
 def _as_dict_safe(
     value: Union[None, Iterable[Any], Mapping[str, Any]]
-) -> Mapping[Any, Any]:
+) -> MutableMapping[Any, Any]:
```

Comment:
Why don't you make this generic so we can restrict the return type?

```
TKey = TypeVar(blabla)
TValue = TypeVar(blabla)

def _as_dict_safe(
    value: Union[None, Iterable[Tuple[TKey, TValue]], Mapping[TKey, TValue]]
) -> MutableMapping[TKey, TValue]
```

This way if you call it with a str str dict, you get a str str dict.

Reply:
updated